test/tests/result_tests.py

```diff
@@ -544,4 +544,43 @@ def test_re_search(self):
         for key, answer in answers.items():
             self.assertEqual(results[key], answer)
 
+    def test_flatten_results(self):
```

Comment:
If this functionality should support updating per_file results as well as a global result, that should be tested here as well.

Reply:
I extended the test.